### PR TITLE
Implement rest of standard API state endpoints

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
Implement standard API state endpoints.
- GetStateFinalityCheckpoints
- GetStateValidatorBalances
- GetStateEpochCommittees
## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateEpochCommittees.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        @OpenApiParam(
+            name = COMMITTEE_INDEX,
+            description = COMMITTEE_INDEX_QUERY_DESCRIPTION,
+            isRepeatable = false),
+        @OpenApiParam(name = SLOT, description = SLOT_QUERY_DESCRIPTION, isRepeatable = false)
+      },
+      responses = {
+        @OpenApiResponse(
+            status = RES_OK,
+            content = @OpenApiContent(from = GetStateRootResponse.class)),
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateEpochCommittees.java]: needs a different response object here or openapi will be incorrect in the schema tag

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java [hunk 2]
```diff
@@ -71,8 +71,7 @@ public GetStateFork(final DataProvider dataProvider, final JsonProvider jsonProv
       method = HttpMethod.GET,
       summary = "Get state fork",
       tags = {TAG_V1_BEACON, TAG_VALIDATOR_REQUIRED},
-      description =
-          "Request beacon node to provide all validators that are scheduled to propose a block in the given epoch.",
+      description = "Returns Fork object for state with given 'stateId'.",
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java]: `state_id`

File: data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java [hunk 3]
```diff
@@ -110,6 +112,48 @@ public GetForkResponse getForkInfo() {
...
+                                        .orElse(true))
+                            .filter(
+                                committee ->
+                                    maybeIndex
+                                        .map(
+                                            index ->
+                                                committee.getCommitteeIndex().intValue() == index)
+                                        .orElse(true))
+                            .map(EpochCommitteeResponse::new)
+                            .collect(toList())));
```
Comment (rolfyone) [inline data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java]: possible to make that a bit easier to digest?

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateEpochCommittees.java (hunk 1 above)
Comment (cemozerr) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateEpochCommittees.java]: Done.

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java (hunk 2 above)
Comment (cemozerr) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java]: Fixed in the other PR, which this builds upon. Hoping to merge that one first.

File: data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java (hunk 3 above)
Comment (cemozerr) [reply data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java]: Donne.

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**